### PR TITLE
Add well known name for Srds stats

### PR DESCRIPTION
**Pull request by DiazAlan**
Commit Message: Add new tag extractor for Srds stats
Additional Description:
Risk Level: Low
Testing: Expanded unit test for srds.
Docs Changes:
Release Notes:
Platform Specific Features:

### CODE REVIEW

File: source/common/config/well_known_names.cc [hunk 1]
```diff
@@ -181,6 +181,10 @@ TagNameValues::TagNameValues() {
   // match.
   addRe2(RDS_ROUTE_CONFIG, R"(^http\.<TAG_VALUE>\.rds\.((<ROUTE_CONFIG_NAME>)\.)\w+?$)", ".rds.");
 
+  // http.[<stat_prefix>.]scoped_rds.(<scoped_route_config_name>.)<base_stat>
+  addRe2(SCOPED_RDS_ROUTE_CONFIG,
+         R"(^http\.<TAG_VALUE>\.scoped_rds\.((<ROUTE_CONFIG_NAME>)\.)\w+?$)", ".scoped_rds.");
```
Comment (adisuissa) [inline source/common/config/well_known_names.cc:186]:
```suggestion
         R"(^http\.<TAG_VALUE>\.scoped_rds\.((<SCOPED_ROUTE_CONFIG_NAME>)\.)\w+?$)", ".scoped_rds.");
```

File: source/common/config/well_known_names.h [hunk 2]
```diff
@@ -143,6 +143,8 @@ class TagNameValues {
   const std::string RESPONSE_CODE_CLASS = "envoy.response_code_class";
   // Route config name for RDS updates
   const std::string RDS_ROUTE_CONFIG = "envoy.rds_route_config";
+  // Scoped route config name for RDS updates
+  const std::string SCOPED_RDS_ROUTE_CONFIG = "envoy.scoped_rds_route_config";
```
Comment (adisuissa) [inline source/common/config/well_known_names.h:147]:
```suggestion
  const std::string SCOPED_RDS_CONFIG = "envoy.scoped_rds_config";
```

File: test/common/stats/tag_extractor_impl_test.cc [hunk 3]
```diff
@@ -398,6 +398,18 @@ TEST(TagExtractorTest, DefaultTagExtractors) {
   regex_tester.testRegex("http.rds_connection_manager.rds.route_config.123.update_success",
                          "http.rds.update_success", {rds_hcm, rds_route_config});
 
+  Tag scoped_rds_hcm;
```
Comment (adisuissa) [inline test/common/stats/tag_extractor_impl_test.cc:401]: nit: add:
```
// SRDS.
```

**Review by adisuissa — COMMENTED** [review]
Thanks for cleaning this up!
Minor nits, but otherwise LGTM.

File: source/common/config/well_known_names.h (hunk 2 above)
Comment (DiazAlan) [reply source/common/config/well_known_names.h]: Done

File: source/common/config/well_known_names.cc (hunk 1 above)
Comment (DiazAlan) [reply source/common/config/well_known_names.cc:189]: Done. Also added associate regex to the top

**Pull request closed without merging**